web-client/src/pages/MasterPage.tsx

```diff
@@ -8,13 +8,25 @@ import {
 import { NewRequestLocation } from 'src/modules/request/pages/routes/NewRequestRoute/constants';
 
 import modules from '../modules';
+import DashboardRoute from './routes/DashboardRoute';
 import NotFoundRoute from './routes/NotFoundRoute';
 import ProtectedRoute from './routes/ProtectedRoute';
 
 const MasterPage = (): ReactElement => {
   const renderModules = () =>
     Object.keys(modules).map(moduleName => {
       const routeModule = modules[moduleName];
+      if (routeModule.layout === 'dashboard') {
+        return (
+          <DashboardRoute
+            key={moduleName}
+            path={routeModule.path}
+            component={routeModule.component}
+            menuItems={routeModule.menuItems || []}
+          />
+        );
+      }
```

Comment:
Same as above

Reply:
If someone copies they dashboard code they won’t see it working correctly unless they also change this code. 

> As we discussed with @ashwinkjoseph this is not required at all because most of our pages will use the same layout except login steps. But I couldn't figure out how to merge DashboardRoute and ProtectedRoute at this moment. That's why I used this for just experimentation. Maybe the DashboardRoute is unnecessary, then we can just use only the ProtectedRoute.

@ashwinkjoseph do have any insights on how to pull this off?

Reply:
yea, I don't think that it should be done in this manner, I am refactoring this a little bit.
the approach is right, but as we discussed, to get this right we have to have changes in the `ProtectedRoute` component as well.
The two should be independent and modules should be able to specify independently whether it should be protected or not irrespective of whether it's having this layout.

Reply:
I don't fully understand what you meant @sabind but I will refactor this a little bit and can you review it after that and let me know if it  solves the issue you mentioned?

Reply:
@sabind can you check this part of the code in https://github.com/reach4help/reach4help/pull/481/commits/3386f42c4314c4fd96aed293445cfc25f11c3370 
would this be fine or do you think we need to make changes to it?
I was trying to do it as children, but then had issues with sending the children to the <Route> as a component prop as the type they were expecting was conflicting with the type for react children

Reply:
Yea I was thinking in other libs/frameworks that these are done with child views and stacked layouts. If that’s not the case then that’s fine, but this does seem weird with a check about which layout should be rendered.

Reply:
Yea, I was trying to do the same but was running into `type` issues as the `type definition` of `children` is not compatible with what the `Route` component expects.
But I have the next best solution for this. I will push this change now